ui/components/Header/index.tsx

```diff
@@ -94,17 +96,72 @@ const JOIN_ROUND_MUTATION = gql`
   }
 `;
 
-const Header = ({ currentUser, fetchingUser, group, round, bucket }) => {
+const SUPER_ADMIN_START = gql`
+  mutation StartSuperAdminSession($duration: Int!) {
+    startSuperAdminSession(duration: $duration) {
+      id
+    }
+  }
+`;
+
+const SUPER_ADMIN_END = gql`
+  mutation EndSuperAdminSession {
+    endSuperAdminSession {
+      id
+    }
+  }
+`;
+
+const GET_SUPER_ADMIN_SESSION = gql`
+  query GetSuperAdminSession {
+    getSuperAdminSession {
+      id
+      duration
+      start
+      end
+    }
+  }
+`;
+
+const Header = ({ currentUser, fetchingUser, group, round, bucket, ss }) => {
   const router = useRouter();
   const [isMenuOpen, setMenuOpen] = useState(false);
   const [editProfileModalOpen, setEditProfileModalOpen] = useState(false);
   const intl = useIntl();
 
+  const [{ data: superAdminSession }] = useQuery({
+    query: GET_SUPER_ADMIN_SESSION,
+  });
   const [, joinGroup] = useMutation(JOIN_GROUP_MUTATION);
   const [, acceptInvitation] = useMutation(ACCEPT_INVITATION);
+  const [, startSuperAdminSession] = useMutation(SUPER_ADMIN_START);
+  const [, endSuperAdminSession] = useMutation(SUPER_ADMIN_END);
 
   const [, joinRound] = useMutation(JOIN_ROUND_MUTATION);
   const color = round?.color ?? "anthracit";
+  const [superAdminTime, setSuperAdminTime] = useState<HTMLElement>();
+  const [inSession, setInSession] = useState(false);
+
+  useEffect(() => {
+    if (ss?.start + ss?.duration * 60000 - Date.now() > 0) {
+      setInSession(true);
+    }
+    if (superAdminTime && superAdminTime) {
```

Comment:
?

Reply:
@alinauroz - this is the timeout, right? Please try to make the variables a little more legible.

Reply:
Not timeout. superAdminTime is the reference of the element which shows superAdmin's remaining time on frontend. But it is only enough to it once.